### ISSUE TITLE
[Bug] Updates screen is flooded when migrating

### ISSUE THREAD

**Issue by scb261** [tachiyomiorg/tachiyomi#4732]
## Device information
* Tachiyomi version: r3185

## Steps to reproduce
1. Clear database in advanced settings to make sure you don't have prefetched chapters from different source.
2. Open a manga in your library.
3. ⋮ -> `Migrate` -> Choose manga from different source -> `Migrate`.
5. Go to the `Updates` screen.

### Expected behavior
See new chapters from different manga and sources.

### Actual behavior
See too many chapters from the manga you just migrated.

## Other details
PR that caused the issue to be more annoying #4645. Before this, chapters where also filtered by upload date, so chapters older than 3 months didn't show up in the Updates. So even when manga is released weekly, it would show about 12 chapters, but now it will show all chapters from the manga. And if you migrate a manga with more than 500 chapters, the Updates screen becomes entirely useless since number of chapters there is limited to 500.

I also tried to fix it myself in the migration code (wanted to transfer fetch date along with read status), but I couldn't, that's why I'm creating this issue.

**Comment by paulo27ms** [comment]
More of an issue with update not sorting correctly... Even without migration, if you update your library after a long time, updated chapters all appear on the same day.
Not necessarily an "issue" even, just a different perspective on how the update tab should work.

**Comment by arkon** [comment]
I've reverted the change from #4645 for now since migration and what @paulo27ms mentioned are more common scenarios than sources that don't have chapter dates.

We'll need to figure out a better way of handling that.

**Comment by paulo27ms** [comment]
Could you not handle those cases by associating fetch date to chapters that don't have dates? In those cases, the fetch dates become their "source date" that other sources have.

**Comment by arkon** [comment]
Yes, we could potentially coalesce it in the query, or set it when creating the chapter objects.

**Comment by paulo27ms** [comment]
@arkon am I misunderstanding the sorting?
After the update it's still sorting by when the chapter was fetched in the update tab (i.e. update library date).
Also didn't fix the flood issue when migrating.
The code says upload date for the query but that's not what's happening...

**Comment by scb261** [comment]
@paulo27ms right now the app queries database by upload date, then sorts it by fetch date in the Updates screen. Preview version had query by fetch date, but it was reverted due to flooding when migrating. I guess 1875c4a (will be in the next release) is supposed to fix it, so maybe query by fetch date will be back. Or as said above it will use fetch date in cases when upload date is missing.

**Comment by paulo27ms** [comment]
I see, so the sorting never changed, I thought it was sorting by how it was querying, which made me confused.

**Comment by rsmyst** [comment]
Uh looks like the issue still persists... Is a fix coming anytime soon? It's getting a little annoying to scroll through a massively long list of chapters in the update screen.

**Comment by stevenyomi** [comment]
Also if you add a title without fetching its chapters (either by long pressing it from the manga list or in the details screen but your connection is too slow) and visit it afterwards, all its chapters will show up in Updates screen. You have to manually remove and readd the title to clear them out.
